neural_compressor/torch/algorithms/smooth_quant/utility.py

```diff
@@ -0,0 +1,2591 @@
+                for o_num in range(num_out):
+                    if pre_out[o_num] == inp:
+                        if cur_op in inplace_ops and (pre_op in ["conv2d", "conv3d", "linear"]):
+                            op_patterns.append([(pre_id, pre_op), (cur_id, cur_op)])
+                        if cur_op in elt_wise and (pre_op in ["conv2d", "conv3d", "linear", "add"]):
+                            op_patterns.append([(pre_id, pre_op), (cur_id, cur_op)])
+                        if cur_op == "add":
+                            pre_ops[i_num] = [pre_id, pre_op]
+        if len(pre_ops) > 0:
+            for key, value in pre_ops.items():
+                if (
+                    value[1] in ["conv2d", "conv3d", "linear"]
+                    and default_cfgs[cur_id]["inputs_quantized"][key] is False
+                ):
+                    op_patterns.append([(value[0], value[1]), (cur_id, cur_op)])
+    return op_patterns
+
+
+def dump_model_op_stats(tune_cfg):
+    """This is a function to dump quantizable ops of model to user.
+
+    Args:
+        tune_cfg (dict): quantization config
+    Returns:
+        None
+    """
+    res = dict()
+    for k, v in tune_cfg["op"].items():
+        op_type_list = k[-1].split("><")
+        op_type = ""
+        for op in op_type_list:
+            if "class" in op:
+                op_type = (
+                    op[op.rfind(".") + 1 : op.rfind("'")]
+                    if op_type == ""
+                    else op_type + "&" + op[op.rfind(".") + 1 : op.rfind("'")]
+                )
+            elif "method" in op:
+                start = op.find("'") + 1
+                if start > 1:
+                    op_type = (
+                        op[start : op.find("'", start)]
+                        if op_type == ""
+                        else op_type + "&" + op[start : op.find("'", start)]
+                    )
+                else:
+                    start = op.find("method") + 7
+                    op_type = (
+                        op[start : op.find(" ", start)]
+                        if op_type == ""
+                        else op_type + "&" + op[start : op.find(" ", start)]
+                    )
+            else:
+                op_type = op if op_type == "" else op_type + "&" + op
+        if op_type not in res.keys():
+            res[op_type] = {"INT8": 0, "BF16": 0, "FP32": 0}
+        if v["weight"]["dtype"] == "int8":
+            res[op_type]["INT8"] += 1
+        elif v["weight"]["dtype"] == "fp32":
+            res[op_type]["FP32"] += 1
+
+    output_data = [
+        [op_type, sum(res[op_type].values()), res[op_type]["INT8"], res[op_type]["BF16"], res[op_type]["FP32"]]
+        for op_type in res.keys()
+    ]
+
+    Statistics(
+        output_data, header="Mixed Precision Statistics", field_names=["Op Type", "Total", "INT8", "BF16", "FP32"]
+    ).print_stat()
+
+
+def get_quantizable_ops_recursively(model, example_inputs):

```

Comment:
I noticed the same function [here](https://github.com/intel/neural-compressor/blob/2a86aeafc754ca3b7495138381efb8faa9397fdf/neural_compressor/torch/algorithms/static_quant/utility.py#L375). If they indeed serve the same purpose, please reuse the code. Also, make sure to review other functions.

Reply:
@violetch24  I saw you moved them to `torch/utils/utility.py`, which is not so reasonable. this utility aims to provide userfacing utilities while these functions are pure for intenal usage and only for IPEX. 
@yiliu30 I prefer to keep them in static_quant folder and import them to smooth_quant folder. Per my understanding, smooth quant is an enhancement of static quant, so we make static quant as the base.